students/tammyd/lesson08/circle.py

```diff
@@ -0,0 +1,71 @@
+"""create a circle using python"""
+
+import math
+
+#step 1: create a class
+class Circle():
+    #the radius is a required parameter 
+    def __init__(self, radius):
+        self.radius = radius
+    def radius(self):
+        radius = self.radius
+        return radius
+
+
+#attribute for the radius
+#c1.radius
+#c2.radius
+
+#c1 = Circle(the_radius)
+
+
+
+#Step 2: add a “diameter” property, so the user can get the diameter
+    @property
+    def diameter(self):
+        return self._radius * 2
+
+#Step 3: Set up the diameter property so that the user can set the diameter of the circle:
+    @diameter.setter
+    #program requires a "def diameter" twice, correct?
+    def diameter(self, value):
+        self._radius = value
+
+#Step 4: Add an area property so the user can get the area of the circle
+    @property
+    def area(self):
+        #A = pi*r^2
+        return math.pi * (self._radius ** 2)
+
+#Step 5: Add an “alternate constructor” that lets the user create a Circle directly with the diameter
+    def from_diameter(diameter):
```

Comment:
This needs the `@classmethod` decoration and needs to accept as its first parameter the `cls` parameter. This will not work as is.